shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -100,13 +102,23 @@ public boolean clipboardHasStrings() {
       };
 
   public PlatformPlugin(Activity activity, PlatformChannel platformChannel) {
+    this.fragmentActivity = null;
     this.activity = activity;
     this.platformChannel = platformChannel;
     this.platformChannel.setPlatformMessageHandler(mPlatformMessageHandler);
 
     mEnabledOverlays = DEFAULT_SYSTEM_UI;
   }
 
+  public PlatformPlugin(FragmentActivity fragmentActivity, PlatformChannel platformChannel) {
+    this.fragmentActivity = fragmentActivity;
```

Comment:
rather than overloading here (which adds extra instance state tracking burden), just check if activity is androidx.activity.ComponentActivity in popSystemNavigator

Reply:
actually checking for androidx.activity.OnBackPressedDispatcherOwner might be even more semantic